dataprofiler/labelers/data_processing.py

```diff
@@ -1129,27 +1146,40 @@ def convert_to_unstructured_format(self, data, labels):
             if start < text_len:
                 entities.append((start - separator_length, start, 'PAD'))
 
-        return text, dict(entities=entities)
+        return text, entities
 
     def process(self, data, labels=None, label_mapping=None, batch_size=32):
         """
         Process structured data for being processed by the
         CharacterLevelCnnModel.
 
         :param data: List of strings to create embeddings for
-        :type data: Union[numpy.ndarray, pandas.DataFrame]
+        :type data: numpy.ndarray
         :param labels: labels for each input character
-        :type labels: list
+        :type labels: numpy.ndarray
         :param label_mapping: maps labels to their encoded integers
         :type label_mapping: Union[dict, None]
         :param batch_size: Number of samples in the batch of data
         :type batch_size: int
         :return batch_data: A dict containing  samples of size batch_size
         :rtype batch_data: dict
         """
-        if labels is not None and not label_mapping:
-            raise ValueError('If `labels` are specified, `label_mapping` must '
-                             'also be specified.')
+        if labels is not None:
+            if not label_mapping:
+                raise ValueError('If `labels` are specified, `label_mapping` '
+                                 'must also be specified.')
+            if data.shape != labels.shape:
+                raise ValueError("Data and labels given to "
+                                 "StructCharPreprocessor have different shapes")
+
+        if len(data.shape) > 1:
+            if not (len(data.shape) == 2 and data.shape[1] == 1):
+                warnings.warn("Data given to StructCharPreprocessor was "
+                              "multidimensional, it will be flattened")
```

Comment:
maybe needs to mention this could affect results, because doesn't say why it needs to be warned, just warning the user. Also maybe add a little more description as in `flattened for model processing`